src/session/inventory_session.go

```diff
@@ -0,0 +1,53 @@
+package session
+
+import (
+	"context"
+	"fmt"
+	"github.com/filanov/bm-inventory/client"
+	"github.com/filanov/bm-inventory/pkg/requestid"
+	"github.com/ori-amizur/introspector/src/config"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"net/url"
+)
+
+func createUrl() string {
+	return fmt.Sprintf("http://%s:%d/%s", config.GlobalConfig.TargetHost, config.GlobalConfig.TargetPort, client.DefaultBasePath)
+}
+
+
+type InventorySession struct {
+	ctx context.Context
+	logger logrus.FieldLogger
+	client *client.BMInventory
+}
+
+func (i *InventorySession) Context() context.Context {
+	return i.ctx
+}
+
+func (i *InventorySession) Logger() logrus.FieldLogger {
```

Comment:
I think you are working too hard with this.
Just initialize your components with the logger,